mytoyota/client.py

```diff
@@ -540,3 +545,62 @@ async def get_trip_json(self, vin: str, trip_id: str) -> str:
         """
         trip = await self.get_trip(vin, trip_id)
         return json.dumps(trip.raw_json, indent=3)
+
+    async def set_lock_vehicle(self, vin: str) -> VehicleLockUnlockActionResponse:
+        """Sends a lock command to the vehicle.
+
+        Args:
+            vin (str): Vehicle identification number.
+
+        Raises:
+            ToyotaLoginError: An error returned when updating token or invalid login information.
+            ToyotaActionNotSupported: The lock action is not supported on this vehicle.
+            ToyotaInternalError: An error occurred when making a request.
+            ToyotaApiError: Toyota's API returned an error.
+        """
+        _LOGGER.debug(f"Locking {censor_vin(vin)}...")
+        raw_response = await self.api.set_lock_unlock_vehicle_endpoint(vin, "lock")
+        _LOGGER.debug(f"Locking {censor_vin(vin)}... {raw_response}")
+        response = VehicleLockUnlockActionResponse(raw_response)
+        return response
+
+    async def set_unlock_vehicle(self, vin: str) -> VehicleLockUnlockActionResponse:
+        """Send an unlock command to the vehicle.
+
+        Args:
+            vin (str): Vehicle identification number.
+
+        Raises:
+            ToyotaLoginError: An error returned when updating token or invalid login information.
+            ToyotaActionNotSupported: The lock action is not supported on this vehicle.
+            ToyotaInternalError: An error occurred when making a request.
+            ToyotaApiError: Toyota's API returned an error.
+        """
+        _LOGGER.debug(f"Unlocking {censor_vin(vin)}...")
+        raw_response = await self.api.set_lock_unlock_vehicle_endpoint(vin, "unlock")
+        _LOGGER.debug(f"Unlocking {censor_vin(vin)}... {raw_response}")
+        response = VehicleLockUnlockActionResponse(raw_response)
+        return response

```

Comment:
@DurgNomis-drol : Sorry, I do not agree. It is fine for me if there is a `set_lock_vehicle` and a `set_unlock_vehicle` function. In my opinion this doesn't have to be a single function where the desired operation is passed as a string argument.

Reply:
I'm also a fan of explicit lock and unlock functions, even the fact that the API exposes a single endpoint for locking and unlocking feels a little off to me.

Reply:
My main grief with this is that i don't want duplicate code, and keeping it now results in too much I believe, if it can be reduced I am okay with keeping it as to separate functions 😃